gear/user_management/src/python/user_app/run.py

```diff
@@ -35,77 +36,111 @@ def read_yaml_file(file_bytes: bytes) -> Optional[List[Any]]:
     return entry_docs[0]
 
 
-# pylint: disable=too-many-locals
-def main() -> None:
-    """Main method to manage users."""
-
-    with GearToolkitContext() as gear_context:
-        gear_context.init_logging()
-        gear_context.log_config()
-
-        default_client = gear_context.client
-        if not default_client:
-            log.error('Flywheel client required to confirm gearbot access')
-            sys.exit(1)
-
-        path_prefix = gear_context.config.get("apikey_path_prefix",
-                                              "/prod/flywheel/gearbot")
-        log.info('Running gearbot with API key from %s/apikey', path_prefix)
+class UserManagementVisitor(GearExecutionEnvironment):
+    """Defines the user management gear."""
+
+    def __init__(self, admin_id: str, client: ClientWrapper,
+                 user_filepath: str, auth_filepath: str):
+        self.__admin_id = admin_id
+        self.__client = client
+        self.__user_filepath = user_filepath
+        self.__auth_filepath = auth_filepath
+
+    @classmethod
+    def create(
+        cls,
+        context: GearToolkitContext,
+        parameter_store: Optional[ParameterStore] = None
+    ) -> 'UserManagementVisitor':
+        """Visits the gear context to gather inputs.
+
+        Args:
+            context (GearToolkitContext): The gear context.
+        """
+        client = ContextClient.create(context=context)
+
```

Comment:
I had changed this intentionally because I don't think it does need to be the gear bot. It's run by a gear rule within nacc/project-admin. I think the permission level should be high enough.
Obviously need to confirm when deploy to production.